src/opnsense/mvc/app/controllers/OPNsense/Base/ApiMutableTableModelControllerBase.php

```diff
@@ -198,21 +212,19 @@ public function toggleItemAction($uuid, $enabled = null)
             if ($uuid != null) {
                 $node = $mdl->getNodeByUUID($uuid);
                 if ($node != null) {
+                    if ($enabled == "0" || $enabled == "1") {
```

Comment:
Toggle needs to happen *before* hook call to be consistent, i.e. model should be in desired state not in current state.